### ISSUE TITLE
Notification light no longer works after Android 7.0 update

### ISSUE THREAD

**Issue by crew5930** [k9mail/k-9#2369]
Please search to check for an existing issue (including closed issues, for which the fix may not have yet been released) before opening a new issue: https://github.com/k9mail/k-9/issues?q=is%3Aissue

### Expected behavior
Tell us what should happen
As new emails come in the notification light based on the color you choose in the setting should flash even when your phone is in sleep mode (not using it and idle).
### Actual behavior
Tell us what happens instead
After the android 7.0 update new emails do show up but no notification light anymore. If I wake my phone I can see the email icon and a preview of the unread emails but no notification light. the only way I can see if I have new emails is to wake the phone
### Steps to reproduce
1. The notification light no longer flashes when new emails come in. I tried turning off the notification and back on still nothing. before the 7.0 update was working fine.
2.
3.

### Environment
K-9 Mail version:
5.203
Android version:
Android 7.0 - Samsung Galaxy S7
Account type (IMAP, POP3, WebDAV/Exchange):
imap

**Comment by daquexian** [comment]
Similar to #2331 ?

**Comment by crew5930** [comment]
No because my sound can be on or off and No blinking light at all. However if I get a Facebook message my phone does blink. But k9 new email it does not. .

**Comment by cketti** [comment]
We're not doing anything special here. Chances are this is some OS weirdness and/or feature.

**Comment by crew5930** [comment]
It appears it is something with the new operating system that clashes with the K9 application. I did find a workaround I notice before my original settings for notifications turned off the sound notification and no vibrate just a blinking light and I was getting notifications prior to the recent Android 7.0 update. Now if I change the notification to either turn on the sound in K9 or do a vibrate then the LED light will blink. if you have the sound and vibrate both turned off in k9 email then the K9 LED blinking light will not work. So for now I have it set to vibrate and vibrate only once as a workaround. It's still an issue though. I should be able to have vibrate and sound turned off in k9 and still get a notification blinking light but it appears with the new Android you have to have one or the other if using the K-9 email.

**Comment by crew5930** [comment]
It appears it is something with the new operating system that clashes with the K9 application. I did find a workaround I notice before my original settings for notifications turned off the sound notification and no vibrate just a blinking light and I was getting notifications prior to the recent Android 7.0 update. Now if I change the notification to either turn on the sound in K9 or do a vibrate then the LED light will blink. if you have the sound and vibrate both turned off in k9 email then the K9 LED blinking light will not work. So for now I have it set to vibrate and vibrate only once as a workaround. It's still an issue though. I should be able to have vibrate and sound turned off in k9 and still get a notification blinking light but it appears with the new Android you have to have one or the other if using the K-9 email. 

----- Original Message -----

**Comment by NIPE-SYSTEMS** [comment]
@cketti I have Android 8 (bit this was also in 7). This issue happens also to me. My phone is a Nexus 5X and I'm running the factory version (which comes directly from Google).

> Chances are this is some OS weirdness and/or feature.

Also on a factory version of Android? This would be unlikely. My guess is that Google has changed the API how an app (K9) must interact with the system to blink the light.

That the "not-blinking" is also present in the version that Google produces (on which all other Android 7 & 8 versions of other vendors are based on) means that currently the blinking will not work on *ALL* Android 7 & 8 devices out there. Or am I wrong? 😉

Please reopen the issue and further investigate it. I would like to help where I can! (e.g. logcat, etc.)

Thank you very much!

**Comment by NIPE-SYSTEMS** [comment]
I've just done a `adb logcat` of my Nexus 5X running on Android 8.

```
10-14 09:49:20.628  2620  2637 D k9      : deleting temp file
10-14 09:49:20.825  2620  2637 W Notification: Use of stream types is deprecated for operations other than volume control
10-14 09:49:20.825  2620  2637 W Notification: See the documentation of setSound() for what to use instead with android.media.AudioAttributes to qualify your playback use case
10-14 09:49:20.896  2620  2637 W Notification: Use of stream types is deprecated for operations other than volume control
10-14 09:49:20.900  2620  2637 W Notification: See the documentation of setSound() for what to use instead with android.media.AudioAttributes to qualify your playback use case
10-14 09:49:20.901  2620  2637 W Notification: Use of stream types is deprecated for operations other than volume control
10-14 09:49:20.901  2620  2637 W Notification: See the documentation of setSound() for what to use instead with android.media.AudioAttributes to qualify your playback use case
```

Based on these logging messages, here are some possible solutions and resources:

https://stackoverflow.com/a/45395800
https://stackoverflow.com/a/45692202

But this only applies to some sound of the notification (which I've disabled on my device, so cannot test that). It has nothing to do with the blinking light.

**Comment by NIPE-SYSTEMS** [comment]
Anyone? Can anyone reopen the issue please? What are the next steps? How can we fix that?

Thank you very much!
